elated-egrets/bangazon-api
# added safe delete policy and hidden pk lookup for payment types

## DESCRIPTION

## How to pull down
Commands

```
git fetch --all
git pull payment-safe-delete
git checkout payment-safe-delete
python manage.py runserver
```

## Steps to test
1. make sure you are in the correct virtual env and in the source directory
1. if you encounter any database related issues, nuke your db and migrations and rerun migration commands
1. cd into root directory of repo
1. run - python manage.py runserver
1. load up localhost port and go to the /api directory
1. make sure you have a user in your database (needed for creating a payment type)
1. create a payment type, ensure that it appears in the list view on /api/payment , as well as /api/payment/(payment_id_here) , make note of the id for the payment type
1. delete the payment type
1. it should still be in the sql database
1. check /api/payment, the deleted payment should no longer be listed
1. go to url for the detail listing for the payment type, it should still be accessible on that url

## What you changed
1. edited payment model and added configuration from safe-delete to make sure that payments are not permanently lost when deleted. 

## What was added to read me
1. as part of this pr I also updated the readme to add more detailed instructions for setting up and entering the virtual environment needed for this project
1. I added more information to the wiki page for payments to reflect its new behavior

## Final Checklist
Before merging.
1. Has each file changed been reviewed?
1. Has the readme been updated to reflect changes made?
1. Has a teammate approved the PR?

Then Ship It!
